libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -553,8 +565,7 @@ private void notEntitled(String message, Class<?> callerClass, String componentN
         // Don't emit a log for muted classes, e.g. classes containing self tests
         if (mutedClasses.contains(callerClass) == false) {
             var moduleName = callerClass.getModule().getName();
-            var loggerSuffix = "." + componentName + "." + ((moduleName == null) ? ALL_UNNAMED : moduleName);
-            var notEntitledLogger = LogManager.getLogger(PolicyManager.class.getName() + loggerSuffix);
+            var notEntitledLogger = getLogger(componentName, moduleName);
```

Comment:
Shouldn't we already have the logger? The caller to  notEntitled should have grabbed ModuleEntitlements?

Reply:
That seems cleaner. Let me try it out.

Reply:
Yeah that's much better.